packages/common/src/services/audius-backend/solana.ts

```diff
@@ -107,66 +125,56 @@ export const createUserBankIfNeeded = async (
     ethAddress ?? audiusLibs.Account!.getCurrentUser()?.wallet
 
   if (!recipientEthAddress) {
-    console.error(
-      "createUserBankIfNeeded: Unexpectedly couldn't get recipient eth address"
+    throw new Error(
+      `createUserBankIfNeeded: Unexpectedly couldn't get recipient eth address`
     )
-    return
   }
 
   try {
-    const res = await audiusLibs.solanaWeb3Manager!.createUserBankIfNeeded({
-      feePayerOverride,
-      ethAddress: recipientEthAddress,
-      mint
-    })
+    const res: CreateUserBankIfNeededResult =
+      await audiusLibs.solanaWeb3Manager!.createUserBankIfNeeded({
+        feePayerOverride,
+        ethAddress: recipientEthAddress,
+        mint
+      })
 
-    // If it already existed, return early
-    if ('didExist' in res && res.didExist) {
-      console.log('Userbank already exists')
-      return
+    if (isCreateUserBankIfNeededError(res)) {
+      // Will catch and log below
+      throw res
     }
 
-    // Otherwise we must have tried to create one
-    console.info(`Userbank doesn't exist, attempted to create...`)
-
-    recordAnalytics({
-      eventName: Name.CREATE_USER_BANK_REQUEST,
-      properties: { mint, recipientEthAddress }
-    })
+    // If it already existed, return early
+    if (res.didExist) {
+      console.log('Userbank already exists')
+    } else {
+      // Otherwise we must have tried to create one
+      console.info(`Userbank doesn't exist, attempted to create...`)
 
-    // Handle error case
-    if ('error' in res) {
-      console.error(
-        `Failed to create userbank, with err: ${res.error}, ${res.errorCode}`
-      )
       recordAnalytics({
-        eventName: Name.CREATE_USER_BANK_FAILURE,
-        properties: {
-          mint,
-          recipientEthAddress,
-          errorCode: res.errorCode,
-          error: (res.error as any).toString()
-        }
+        eventName: Name.CREATE_USER_BANK_REQUEST,
+        properties: { mint, recipientEthAddress }
       })
-      return
     }
 
-    // Handle success case
-    console.log(`Successfully created userbank!`)
     recordAnalytics({
       eventName: Name.CREATE_USER_BANK_SUCCESS,
       properties: { mint, recipientEthAddress }
     })
-  } catch (err) {
+    return res.userbank
+  } catch (err: any) {
+    // Catching error here for analytics purposes
+    const errorMessage = 'error' in err ? err.error : (err as any).toString()
+    const errorCode = 'errorCode' in err ? err.errorCode : undefined
```

Comment:
this would be a solana error code i imagine?

Reply:
There are a few error cases that can lead here:
* TransactionHandler in libs gets an error back from relay with a http status code (usually 500). That would be this error code
* An unexpected error while making the http request to relay
* A runtime error of some sort (unexpected empty object, etc).